tools/src/scie_pants/pants_version.py

```diff
@@ -148,9 +148,16 @@ def determine_latest_stable_version(
     ptex: Ptex, pants_config: Path, find_links_dir: Path, github_api_bearer_token: str | None = None
 ) -> tuple[Callable[[], None], ResolveInfo]:
     info(f"Fetching latest stable Pants version since none is configured")
-    pants_version = ptex.fetch_json("https://pypi.org/pypi/pantsbuild.pants/json")["info"][
-        "version"
-    ]
+
+    latest_tag = ptex.fetch_json(
+        "https://github.com/pantsbuild/pants/releases/latest", Accept="application/json"
+    )["tag_name"]
+    if not latest_tag.startswith("release_"):
+        fatal(
+            f'Expected the GitHub Release tagged "latest" to have the "release_" prefix. Got "{latest_tag}"'
```

Comment:
Messages like these are potentially user facing, so would prefer it included a little more context and/or suggested next course of action. (like, this is not you it's a pants bug, please report; that sort of thing.)